torch/_inductor/runtime/triton_helpers.py

```diff
@@ -198,25 +198,71 @@ def any(a, dim):
 
 @triton.jit
 def bucketize_binary_search(
-    values,  # 1D tensor
-    offsets_ptr,
-    indexing_dtype,
-    right,  # bool: if true, use intervals closed on the left; see [Note: Inductor bucketize op]
-    OFFSETS_SIZE: int,
-    BLOCK_SHAPE,  # tuple/list of block shape
+    values: tl.tensor,
+    boundaries_ptr: tl.tensor,
+    BOUNDARIES_SIZE: tl.constexpr,
+    BOUNDARIES_UNDERLYING_NUMEL: tl.constexpr,
+    BOUNDARIES_STRIDE: tl.constexpr,
```

Comment:
should these be constexpr?

afaict, they are not being treated as constexpr. I did a local test, and I see dynamic shapes being passed in if the offsets size changes. And my test is not failing, which I think means that these are not actually being handled as constexpr.

do you think you could remove `constexpr` if it's not needed here? And if it _is_ needed - then can you try testing to see if things work as expected?

(same applies to the other tl.constexprs below in this function)

Reply:
@davidberard98 I'll remove the constexpr, and see if I can get a more appropriate annotation.

Reply:
@davidberard98 Done, and rebased on the latest `viable/strict`. Should be ready to go for you to import!